python/sglang/srt/layers/moe/fused_moe_triton/layer.py

```diff
@@ -1037,6 +1038,8 @@ def forward(self, hidden_states: torch.Tensor, topk_output: TopKOutput):
 
         router_logits = router_logits.to(torch.float32)
 
+        with use_symmetric_memory(get_tp_group(), disabled=not is_dp_max_padding()):
+            symm_output = torch.empty_like(hidden_states)
```

Comment:
when use_symmetric_memory is not enabled, this created a useless tensor.
The principal is: when symmetric_memory is not enabled, the added overhead in the model forward pass should just be < 2 if/else and function calls.

Reply:
There is no overhead here.
When you pass `output=None` or non `output` to `trtllm_fp4_block_scale_moe`, it will call internally `torch.empty_like(hidden_states)`.
So we end up with the exact same behavior.